delivery_postlogistics/models/delivery_carrier.py

```diff
@@ -0,0 +1,168 @@
+# Copyright 2013-2016 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+from odoo.exceptions import UserError
+
+from ..postlogistics.web_service import PostlogisticsWebService
+
+
+class DeliveryCarrier(models.Model):
+    _inherit = "delivery.carrier"
+
+    delivery_type = fields.Selection(
+        selection_add=[("postlogistics", "PostLogistics")],
+        ondelete={"postlogistics": "set default"},
+    )
+    postlogistics_default_package_type_id = fields.Many2one(
+        "stock.package.type", domain=[("package_carrier_type", "=", "postlogistics")]
+    )
+
+    postlogistics_endpoint_url = fields.Char(
+        string="Endpoint URL",
+        default="https://wedecint.post.ch/",
+        required=True,
+    )
+    postlogistics_client_id = fields.Char(
+        string="Client ID", groups="base.group_system"
+    )
+    postlogistics_client_secret = fields.Char(
+        string="Client Secret", groups="base.group_system"
+    )
+    postlogistics_logo = fields.Binary(
+        string="Company Logo on Post labels",
+        help="Optional company logo to show on label.\n"
+        "If using an image / logo, please note the following:\n"
+        "– Image width: 47 mm\n"
+        "– Image height: 25 mm\n"
+        "– File size: max. 30 kb\n"
+        "– File format: GIF or PNG\n"
+        "– Colour table: indexed colours, max. 200 colours\n"
+        "– The logo will be printed rotated counter-clockwise by 90°"
+        "\n"
+        "We recommend using a black and white logo for printing in "
+        " the ZPL2 format.",
+    )
+    postlogistics_office = fields.Char(
+        string="Domicile Post office",
+        help="Post office which will receive the shipped goods",
+    )
+
+    postlogistics_label_layout = fields.Many2one(
+        comodel_name="postlogistics.delivery.carrier.template.option",
+        string="Label layout",
+        domain=[("postlogistics_type", "=", "label_layout")],
+    )
+    postlogistics_output_format = fields.Many2one(
+        comodel_name="postlogistics.delivery.carrier.template.option",
+        string="Output format",
+        domain=[("postlogistics_type", "=", "output_format")],
+    )
+    postlogistics_resolution = fields.Many2one(
+        comodel_name="postlogistics.delivery.carrier.template.option",
+        string="Resolution",
+        domain=[("postlogistics_type", "=", "resolution")],
+    )
```

Comment:
We should stick to standard by using and extending (if needed) the model "report.paperformat" from base module.
https://github.com/odoo/odoo/blob/2efab4e7b90d36ce60c9858576922b504509993a/odoo/addons/base/models/report_paperformat.py#L167